File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    for i in range(len(s)) :
        tmp = True
        for j in range(i - 1, -1, -1) :
            if s[i] == s[j] :
                answer.append(i - j)
                tmp = False
                break
        if tmp :
            answer.append(-1)
    return answer

File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    for (int i = 0; i < s.length(); i++) {
        bool first = true;
        for (int j = i - 1; j > -1; j--) {
            if (s[i] == s[j]) {
                first = false;
                answer.push_back(i - j);
                break;
            }
        }
        if (first) answer.push_back(-1);
    }
    return answer;
}